modules/core/src/utils/url-utils.ts

```diff
@@ -108,31 +107,38 @@ export class URLUtils {
      * Please refer specification for other part implementations of the url:
      * https://www.ietf.org/rfc/rfc2396.txt
      *
-     * @param url {string} a valid url string.
+     * @param url - a valid url string.
      * @returns URLComponentsInterface
      * @throws Invalid URL | undefined
      */
     public static urlComponents(url: string): URLComponentsInterface {
         try {
-            const details = new URL(url.trim());
-            const protocol = details.protocol.replace(":", "");
+            const details:URL = new URL(url.trim());
+            const protocol:string = details.protocol.replace(":", "");
+
             return {
-                protocol, // https|http
+                _url: details, // URL Instance
                 host: details.host, // localhost:9443
-                origin: details.origin, // https://localhost:9443
                 href: details.href, // https://localhost:9443/some/long/url
-                _url: details, // URL Instance
+                origin: details.origin, // https://localhost:9443
                 pathWithoutProtocol: details.href.split("://")[ 1 ],
+                protocol // https|http
             } as URLComponentsInterface;
         } catch (error) {
             return null;
         }
     }
 
     public static isURLValid(url: string, checkForSanity?: boolean): boolean {
+        // List of invalid keywords that cannot be present in a URL.
+        const invalidKeywords: string[] = [ "javascript:" ];
```

Comment:
sanitizeUrl method already handles this, doesn't it?

Reply:
It does not handle these cases

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/5611/commits/c8443438e56f45c40c02e94398280194200ed540